src/lib/parameters/px4params/xmlstringsout.py

```diff
@@ -0,0 +1,129 @@
+#
+# Export all parameter strings for translation into key/value file.
+#
+
+import xml.etree.ElementTree as ET
+import codecs
+import os
+
+def indent(elem, level=0):
+    i = "\n" + level*"  "
+    if len(elem):
+        if not elem.text or not elem.text.strip():
+            elem.text = i + "  "
+        if not elem.tail or not elem.tail.strip():
+            elem.tail = i
+        for elem in elem:
+            indent(elem, level+1)
+        if not elem.tail or not elem.tail.strip():
+            elem.tail = i
+    else:
+        if level and (not elem.tail or not elem.tail.strip()):
+            elem.tail = i
+
+class XMLOutput():
+
+    def __init__(self, groups, board, inject_xml_file_name):
+        unique_strings = set()
+        xml_parameters = ET.Element("properties")
+
+        for group in groups:
+            group_name=group.GetName()
+            if group_name not in unique_strings:
```

Comment:
```py
            if group_name not in unique_strings:
```
Oh, looks like it is

Reply:
Indeed :-)

This is waiting on @dagar - he will review and merge (if appropriate) when it gets near the top of his priority list.